src/wazuh_db/wdb_agent.c

```diff
@@ -664,140 +576,129 @@ int wdb_update_agent_multi_group(int id, char *group) {
 
 /* Find group by name. Returns id if success or -1 on failure. */
 int wdb_find_group(const char *name) {
-    sqlite3_stmt *stmt = NULL;
-    int result;
+    int result = 0;
+    char wdbquery[OS_BUFFER_SIZE] = "";
+    char wdboutput[OS_BUFFER_SIZE] = "";
+    int wdb_sock = -1;
+    char *json_string = NULL;
+    cJSON *root = NULL;
+    cJSON *elem = NULL;
+    cJSON *json_name = NULL;
 
-    if (wdb_open_global() < 0)
-        return -1;
+    snprintf(wdbquery, OS_BUFFER_SIZE, global_db_queries[SQL_FIND_GROUP], name);
+    wdbc_query_ex(&wdb_sock, wdbquery, wdboutput, sizeof(wdboutput));
 
-    if (wdb_prepare(wdb_global, SQL_FIND_GROUP, -1, &stmt, NULL)) {
-        mdebug1("SQLite: %s", sqlite3_errmsg(wdb_global));
-        wdb_close_global();
+    if(json_string = wstr_chr(wdboutput, ' '), json_string ){
+        *json_string='\0';
+        json_string++;
+    } else{
+        mdebug1("SQLite result has no space. Query: %s",wdbquery);
         return -1;
     }
 
-    sqlite3_bind_text(stmt, 1, name, -1, NULL);
-
-    result = wdb_step(stmt) == SQLITE_ROW ? sqlite3_column_int(stmt, 0) : -1;
-    sqlite3_finalize(stmt);
+    if(strcmp(wdboutput,"ok") == 0 && strcmp(json_string,"[]") != 0){
+        root = cJSON_Parse(json_string);    
+        elem = cJSON_GetArrayItem(root, 0);
+        json_name = cJSON_GetObjectItem(elem, "id");
+        result = json_name->valueint;
+    } else{
+        mdebug1("SQLite Query failed: %s", wdbquery);
+        return -1;
+    }
+        
     return result;
 }
 
 /* Insert a new group. Returns id if success or -1 on failure. */
 int wdb_insert_group(const char *name) {
-    sqlite3_stmt *stmt = NULL;
-    int result;
-
-    if (wdb_open_global() < 0)
-        return -1;
-
-    if (wdb_prepare(wdb_global, SQL_INSERT_AGENT_GROUP, -1, &stmt, NULL)) {
-        mdebug1("SQLite: %s", sqlite3_errmsg(wdb_global));
-        wdb_close_global();
-        return -1;
-    }
+    int result = 0;
+    char wdbquery[OS_BUFFER_SIZE] = "";
+    char wdboutput[OS_BUFFER_SIZE] = "";
+    int wdb_sock = -1;
 
-    sqlite3_bind_text(stmt, 1, name, -1, NULL);
+    snprintf(wdbquery, OS_BUFFER_SIZE, global_db_queries[SQL_INSERT_AGENT_GROUP], name);
+    result = wdbc_query_ex(&wdb_sock, wdbquery, wdboutput, sizeof(wdboutput));
 
-    if (wdb_step(stmt) == SQLITE_DONE)
-        result = (int)sqlite3_last_insert_rowid(wdb_global);
+    if (result == 0)
+        result = wdb_find_group(name);
     else {
-        mdebug1("SQLite: %s", sqlite3_errmsg(wdb_global));
+        mdebug1("SQLite query error: %s", wdbquery);
```

Comment:
Make the log explicit according to the error (`-1` or `-2`).

Reply:
Now the -1 Error in the response from socket case is considered in the OS_INVALID switch case. But the -2 Error in the communication. is different from #define OS_NOTFOUND -2 /* Entry not found */